### PR TITLE
Convert PreferenceLoader related classes to Kotlin

### PR DESCRIPTION
**Pull request by cooltey**

### CODE REVIEW

File: app/src/main/java/org/wikipedia/settings/DeveloperSettingsActivity.kt [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import org.wikipedia.activity.SingleFragmentActivity
+
+class DeveloperSettingsActivity : SingleFragmentActivity<DeveloperSettingsFragment>() {
+    public override fun createFragment(): DeveloperSettingsFragment {
+        return DeveloperSettingsFragment.newInstance()
+    }
+
+    companion object {
+        @JvmStatic
+        fun newIntent(context: Context?): Intent {
```
Comment (dbrant) [inline app/src/main/java/org/wikipedia/settings/DeveloperSettingsActivity.kt:14]: Any reason for `context` to be nullable?

File: app/src/main/java/org/wikipedia/settings/DeveloperSettingsPreferenceLoader.kt [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+        true
+    }
+
+    override fun loadPreferences() {
+        loadPreferences(R.xml.developer_preferences)
+        setUpMediaWikiSettings()
+        findPreference(R.string.preferences_developer_crash_key).onPreferenceClickListener = Preference.OnPreferenceClickListener { throw TestException("User tested crash functionality.") }
+        findPreference(R.string.preference_key_add_articles).onPreferenceChangeListener = Preference.OnPreferenceChangeListener { _: Preference, newValue: Any ->
+            if (!isEmptyOrZero(newValue)) {
+                createTestReadingList(TEXT_OF_TEST_READING_LIST, 1, StringUtil.trim(newValue).toInt())
```
Comment (dbrant) [inline app/src/main/java/org/wikipedia/settings/DeveloperSettingsPreferenceLoader.kt:44]: I'm not sure I see why we need our own `trim()` function. (Why can't we use `.toString().trim()`?)

**Review by dbrant — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/wikipedia/settings/DeveloperSettingsActivity.kt (hunk 1 above)
Comment (cooltey) [reply app/src/main/java/org/wikipedia/settings/DeveloperSettingsActivity.kt]: No. Done.

File: app/src/main/java/org/wikipedia/settings/DeveloperSettingsPreferenceLoader.kt (hunk 2 above)
Comment (cooltey) [reply app/src/main/java/org/wikipedia/settings/DeveloperSettingsPreferenceLoader.kt]: Make sense. Done.

**Pull request merged**